### PR TITLE
Clarify lib purpose in docs, make propagate and pre-commit commit the fixes to the issues

### PR DESCRIPTION
**Pull request by SimonBiggs**

### CODE REVIEW

File: .github/workflows/library.yml [hunk 1]
```diff
@@ -347,10 +341,25 @@ jobs:
           git config --local user.email "propagate-bot@pymedphys.com"
           git config --local user.name "PyMedPhys Propagate Bot"
           git add poetry.lock setup.py
-          git commit -m "dummy commit"
+          git commit -m "commit changes to poetry.lock and setup.py"
 
           git status
           git diff-index --quiet HEAD --
+          
+      - name: Fix any issues
```
Comment (Matthew-Jennings) [inline .github/workflows/library.yml:352]: Nice addition!

Comment (Matthew-Jennings) [reply .github/workflows/library.yml:352]: Lemme know when it's working and I can review the job :)

Comment (SimonBiggs) [reply .github/workflows/library.yml:364]: Good to go 🙂

File: .github/workflows/library.yml [hunk 2]
```diff
@@ -77,6 +73,8 @@ jobs:
             task: 'slow'
           - os: 'windows'
             task: 'pyright'
+          - os: 'windows'
```
Comment (Matthew-Jennings) [inline .github/workflows/library.yml:76]: no propagate for ubuntu?

**Review by Matthew-Jennings — COMMENTED** [review]
Sorry mate, probably won't get to this tonight. Pretty happy to trust you if you're happy though. I couldn't find a clear example of where propagate ran successfully when needed. But I did see one where it checked whether it was needed and successfully determined it wasn't.

File: .github/workflows/library.yml (hunk 2 above)
Comment (SimonBiggs) [reply .github/workflows/library.yml:76]: Actually that's the "ignore" list :)... that essentially means only Ubuntu.

**Comment by SimonBiggs** [comment]
> Sorry mate, probably won't get to this tonight.

All good no rush 🙂, happy for it to wait till Monday.

**Review by Matthew-Jennings — APPROVED** [review]
Very cool :)

**Comment by Matthew-Jennings** [comment]
What's the merge protocol? Would you like to?

**Comment by SimonBiggs** [comment]
> What's the merge protocol? Would you like to?

Nothing strict I tend to let the PR creator do the merging.

**Pull request merged**